util/hint/hinterror.go

```diff
@@ -0,0 +1,74 @@
+package hint
+
+import (
+	"errors"
+	"fmt"
+	"regexp"
+	"strings"
+
+	"github.com/earthly/earthly/util/stringutil"
+)
+
+// note this regex should be updated in case the error format changes in Error
+var regex = regexp.MustCompile(`(?P<error>.+?):Hint: (?P<hint>(?s).+)`)
```

Comment:
Nit: `regex` is a bit vague. Should the name be clarified?

Reply:
done